packages/core/integration-tests/test/cache.js

```diff
@@ -6154,6 +6159,75 @@ describe('cache', function () {
       assert.equal(res, 4);
     });
 
+    it('should write bundle graph to cache on bundling error', async function () {
+      let overlayFSPackageManager = new NodePackageManager(
+        overlayFS,
+        __dirname,
+      );
+      let entries = 'source/index.js';
+      let options = {
+        mode: 'production',
+        defaultTargetOptions: {
+          shouldScopeHoist: false,
+        },
+        packageManager: overlayFSPackageManager,
+        shouldDisableCache: false,
+        inputFS: overlayFS,
+        cacheDir: path.join(__dirname, '.parcel-cache'),
+      };
+
+      await fsFixture(overlayFS)`
+      source
+        foo.js:
+
+          export default 2;
+        index.js:
+          import('./foo');
+
+          export default 1;
+        .parcelrc:
+          {
+            "extends": "@parcel/config-default",
+            "bundler": "./test-bundler.js"
+          }
+        test-bundler.js:
+          import {Bundler} from '@parcel/plugin'
+          import DefaultBundler from '@parcel/bundler-default'
+
+          const CONFIG = Symbol.for('parcel-plugin-config');
+
+          export default new Bundler({
+            loadConfig({config, options}) {
+              return DefaultBundler[CONFIG].loadConfig({config, options});
+            },
+          
+            bundle({bundleGraph, config}) {
+              DefaultBundler[CONFIG].bundle({bundleGraph, config});
+            },
+            optimize() {throw new Error("Intentionally throw error")},
+          });
+        yarn.lock:`;
+      // $FlowFixMe
+      await assert.rejects(() => bundle(entries, options));
+
+      let resolvedOptions = await resolveOptions(
+        getParcelOptions(entries, options),
+      );
+
+      let bundleGraphCacheKey = hashString(
+        `${version}:BundleGraph:${
+          JSON.stringify(resolvedOptions.entries) ?? ''
+        }${resolvedOptions.mode}`,
+      );
+
+      let bundleGraphPath = path.join(
+        resolvedOptions.cacheDir,
+        bundleGraphCacheKey + '-0',
```

Comment:
Not sure if this is the best way to determine the bundleGraphPath. Currently being determined [here ](https://github.com/parcel-bundler/parcel/blob/817c7786ef57bfd4810fbcb791c2eb90c78bcbc2/packages/core/cache/src/FSCache.js#L110). Which means if this changes, this code will break.. but I have no way of accessing this function directly.

Reply:
Maybe `getFilePath()` should be added to the Cache interface? /cc @JakeLane

Reply:
Yeah I think that would be reasonable.

Though I think it would be more maintainable for the integration test if we used the cache interface to load the blob instead, as we're duplicating business logic here.

Reply:
Is the goal of the this change to be able to access the graphs via parcel-query? You could use that in the test

Reply:
@mischnic For now we're just looking to confirm that the bundleGraph is written to cache if an error occurs during bundling w/ the ultimate goal of being able to replicate errors from the parcel cache.
@JakeLane I made a change based on what I thought you meant, lmk if you meant something else!